pkg/kubelet/prober/prober_manager.go

```diff
@@ -261,6 +249,20 @@ func (m *manager) UpdatePodStatus(podUID types.UID, podStatus *v1.PodStatus) {
 			started = !exists
 		}
 		podStatus.ContainerStatuses[i].Started = &started
+
+		if started {
```

Comment:
If there is no way container can be ready, but not started, perhaps the status can be stored in a single variable instead of two independent flags?

Reply:
No, there are more ramifications... I think it's clearer to keep both even just for the sake of explanation and ease of testing.

Reply:
Do you mean in the API or in this function?

The API has shipped, we don't want to change that.

This commit was designed to be surgical - move the code block, add an `if`.  Further cleanup may be possible, but it seems low-value to me - this code is pretty simple to read?

Reply:
I meant API =). I commented before looked into it and understand it's seems to be hard to change. The only benefit is guaranteed consistency of the status. This method is ok. But there are more. For instance, this: https://github.com/kubernetes/kubernetes/blob/323f34858de18b862d43c40b2cced65ad8e24052/pkg/kubelet/status/status_manager.go#L241 where `Ready` and `Started` are set seemingly independently.

Reply:
That case (I think) is after readiness has been considered at a lower level, but I admit I am not 100% confident in this code area any more :(